### PR TITLE
JMS: Java API for JmsMessages withProperties

### PR DESCRIPTION
**Pull request by dotbg**
Implementation for #1900

### CODE REVIEW

**Comment by ennru** [comment]
Thank you for adding this. The compiler is not happy with your use of the Java Hashmap.

**Comment by dotbg** [comment]
Fixed

**Comment by ennru** [comment]
Failure was https://github.com/akka/alpakka/issues/1306
https://travis-ci.com/akka/alpakka/jobs/229437607#L1030

File: jms/src/main/scala/akka/stream/alpakka/jms/JmsMessages.scala [hunk 1]
```diff
@@ -219,6 +230,12 @@ final class JmsByteMessage private (bytes: Array[Byte],
   override def withProperties(props: Map[String, Any]): JmsByteMessage =
     copy(properties = properties ++ props)
 
+  /**
+   * Java API.
+   */
+  override def withProperties(map: util.Map[String, Object]): JmsByteMessage =
```
Comment (ennru) [inline jms/src/main/scala/akka/stream/alpakka/jms/JmsMessages.scala]:
```suggestion
  override def withProperties(map: java.util.Map[String, Object]): JmsByteMessage =
```

Comment (dotbg) [reply jms/src/main/scala/akka/stream/alpakka/jms/JmsMessages.scala]: updated

**Comment by ennru** [comment]
Thank you!

**Comment by ennru** [comment]
Backported to 1.1.2 in https://github.com/akka/alpakka/pull/1977

**Pull request merged**